bootstrap/kubeadm/internal/cloudinit/kubeadm-bootstrap-script.sh

```diff
@@ -128,10 +128,8 @@ retry-command kubeadm join phase control-plane-prepare kubeconfig
 retry-command kubeadm join phase control-plane-prepare control-plane
 # {{ end }}
 retry-command kubeadm join phase kubelet-start
-# {{ if .ControlPlane }}
-try-or-die-command kubeadm join phase control-plane-join etcd
-retry-command kubeadm join phase control-plane-join update-status
-retry-command kubeadm join phase control-plane-join mark-control-plane
-# {{ end }}
+
+# Run kubeadm join and skip all already executed phases.
+try-or-die-command kubeadm join --skip-phases preflight,control-plane-prepare,kubelet-start
```

Comment:
We have two options here:

**Current implementation**

Just run `kubeadm join` without any retries at this point.
The disadvantage is that we do not run `control-plane-join mark-control-plane` with retries anymore.
The advantage here (and requirement when `ControlPlaneKubeletLocalMode` is enabled): we also take advantage of hidden and new phases to be run when they are enabled.

**The alternative would be:**

* Keep the current implementation here for worker nodes.
* For ControlPlane nodes: use the `/etc/kubernetes/admin.conf` to check if the `ControlPlaneKubeletLocalMode` is enabled
  * Only if it is **not enabled**: run the `control-plane-join mark-control-plane` with retries as before

I think the advantages of the current implementation are better then doing the alternative here.

Reply:
i think given all recent versions of kubeadm in support by CAPI have integrated retries. breaking down the join in phases and/or retrying any of them is pointless.

Reply:
Agreed. 
At this point we need to fix this script for 1.31 as a stop gap before removal of the entire experimentalRetryMachines, but it does not add any value on top of retry/timeout management embedded in recent versions of kubeadm